music_assistant/providers/bbc_sounds/__init__.py

```diff
@@ -0,0 +1,865 @@
+"""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import AsyncGenerator
+from datetime import timedelta
+from typing import TYPE_CHECKING, cast
+
+from music_assistant_models.config_entries import (
+    ConfigEntry,
+    ConfigValueOption,
+    ConfigValueType,
+    ProviderConfig,
+)
+from music_assistant_models.enums import ConfigEntryType, ImageType, MediaType, ProviderFeature
+from music_assistant_models.errors import LoginFailed, MusicAssistantError
+from music_assistant_models.media_items import (
+    BrowseFolder,
+    ItemMapping,
+    MediaItemImage,
+    MediaItemMetadata,
+    MediaItemType,
+    Podcast,
+    PodcastEpisode,
+    ProviderMapping,
+    Radio,
+    RecommendationFolder,
+    SearchResults,
+    Track,
+)
+from music_assistant_models.unique_list import UniqueList
+
+import music_assistant.helpers.datetime as dt
+from music_assistant.constants import CONF_PASSWORD, CONF_USERNAME
+from music_assistant.controllers.cache import use_cache
+from music_assistant.helpers.datetime import LOCAL_TIMEZONE
+from music_assistant.models.music_provider import MusicProvider
+from music_assistant.providers.bbc_sounds.adaptor import Adaptor
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from music_assistant_models.provider import ProviderManifest
+    from music_assistant_models.streamdetails import StreamDetails
+    from sounds.models import SoundsTypes
+
+    from music_assistant.mass import MusicAssistant
+    from music_assistant.models import ProviderInstanceType
+
+from sounds import (
+    Container,
+    LiveStation,
+    Menu,
+    MenuRecommendationOptions,
+    PlayStatus,
+    SoundsClient,
+    exceptions,
+)
+
+SUPPORTED_FEATURES = {
+    ProviderFeature.BROWSE,
+    ProviderFeature.RECOMMENDATIONS,
+    ProviderFeature.SEARCH,
+}
+
+FEATURES = {"catchup_segments": False, "check_blank_image": False}
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Create new provider instance."""
+    instance = BBCSoundsProvider(mass, manifest, config)
+    await instance.handle_async_init()
+    return instance
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # ruff: noqa: ARG001
+
+    return (
+        ConfigEntry(
+            key=_Constants.CONF_ENABLE_UK_CONTENT,
+            type=ConfigEntryType.BOOLEAN,
+            label="Enable UK Sounds content (beta)",
```

Comment:
```suggestion
            label="Enable UK Sounds content",
```

The whole provider is beta :-)